kolibri/plugins/coach/api.py

```diff
@@ -190,9 +195,12 @@ def list(self, request, *args, **kwargs):
         are requesting notifications in the last five minutes
         """
         # Use super on the parent class to prevent an infinite recursion.
-        response = super(viewsets.ReadOnlyModelViewSet, self).list(
-            request, *args, **kwargs
-        )
+        try:
+            response = super(viewsets.ReadOnlyModelViewSet, self).list(
+                request, *args, **kwargs
+            )
+        except DatabaseError:
+            repair_sqlite_db(connections["notifications_db"])
```

Comment:
similarly, do you think we should return an empty response here?

Reply:
No, here the function does not returns, it has to continue, anyway I have seen another kind of error if the db file does not even exist and has added it to the exceptions.